### PR TITLE
Add confidenceMethod to table of reserved terms and v2 `@context`

### PR DESCRIPTION
**Pull request by OR13**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1142.html" title="Last updated on Aug 2, 2023, 5:30 PM UTC (a0b38f9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1142/e6e9fbf...a0b38f9.html" title="Last updated on Aug 2, 2023, 5:30 PM UTC (a0b38f9)">Diff</a>

### CODE REVIEW

**Comment by OR13** [comment]
I am requesting changes on my own PR, until the links are not 404.

File: contexts/credentials/v2
Comment (OR13) [inline contexts/credentials/v2:24]:
```suggestion
        },
```

File: index.html
Comment (msporny) [reply index.html:3351]: @mprorock @man4prez @kwlinson Do we have approval to move vc-render-method into CCG? That would address this issue.

**Review by brentzundel — CHANGES_REQUESTED** [review]
also requesting changes until the links are valid

File: contexts/credentials/v2
Comment (jandrieu) [inline contexts/credentials/v2:66]:
```suggestion
        "confidenceMethod": {
          "@id": "https://www.w3.org/2018/credentials#confidenceMethod",
          "@type": "@id"
```

Shouldn't this be scoped to credentialSubject?

Comment (dlongley) [reply contexts/credentials/v2:66]: It should either be globally scoped or use `"@propagate": true`, but, yes, making it a property on a VC only isn't correct.

Comment (TallTed) [reply contexts/credentials/v2:66]: I believe `confidenceMethod` could be scoped to any property/ies. I don't understand what it would do for `credentialSubject`, nor what it does for `VerifiableCredential`.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-06-14-vcwg#section1-2) on 2023-06-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.2. Update reservation table #1142 (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._


**Orie Steele:** 1142 reserves the confidenceMethod in the JSON-LD Context and adds the confidence method as a JSON member and adds same at-risk reservation dependency. I added a link that we expect to not be 404. Under render, I attempted to link to document that is furthest along wrt. renderMethod. Not transferred to CCG yet. This one adds links and at risk ffeature can read specs.  

_See github pull request [vc-data-model#1140](https://github.com/w3c/vc-data-model/pull/1140)._




**Manu Sporny:** What's 1140 about?  

**Orie Steele:** Pull 1141 adds an issue marker about debating this particular property and whether we will reserve these things in the core context. This is saying same thing as reserved properties, focuses on property in core context. If we merge other PR, then this issue marker would still remain and at some point confidenceMethod could be removed. That would be the reason to leave issue marker in... never anything documented for this extension point.  

</details>

File: index.html
Comment (msporny) [reply index.html:3351]: @mprorock @man4prez @kwlinson second ping... we need an answer as it's holding up PRs in VCWG.

**Comment by msporny** [comment]
This PR is currently blocked because the links to the CCG specifications do not resolve. In order to fix the links:

1. The CCG Chairs will have to approve the Render Methods work item.
2. The Editor's of the `confidenceMethod` specification will have to propose it as a CCG work item.

File: index.html
Comment (msporny) [reply index.html:3351]: This has now been resolved: https://w3c-ccg.github.io/vc-render-method/

Comment (OR13) [reply index.html:3351]: I'm going to resolve this thread, since the link has been fixed.

File: contexts/credentials/v2
Comment (OR13) [reply contexts/credentials/v2:66]: Based on the comments above, I don't think it should be scoped... However, if it should be scoped, there should be normative statements that people can read, before the context scoping is applied... and some human readable summary of why, here:

https://www.w3.org/2018/credentials#confidenceMethod

**Comment by OR13** [comment]
> The Editor's of the `confidenceMethod` specification will have to propose it as a CCG work item.

^ this is still blocking the PR.

File: contexts/credentials/v2
Comment (OR13) [reply contexts/credentials/v2:66]: Maybe https://w3id.org/security#confidenceMethod would be a better term definition, if the W3C CCG will define the behavior.

**Comment by awoie** [comment]
> > The Editor's of the `confidenceMethod` specification will have to propose it as a CCG work item.
> 
> ^ this is still blocking the PR.

I'll propose this as a work item in the W3C CCG. Give me time until the end of this week.

**Comment by OR13** [comment]
@awoie any update?

**Comment by awoie** [comment]
> @awoie any update?

I proposed a new CCG work item here: https://github.com/w3c-ccg/community/issues/245.

**Review by awoie — APPROVED** [review]
lgtm

File: index.html
Comment (TallTed) [inline index.html:3298]:
```suggestion
might use to increase their confidence that the value of an attribute in or of
a verifiable credential or verifiable presentation is accurate, including but not
limited to attributes such `initialRecipient` (a/k/a `issuee`), `presenter`, `authorizedPresenter`, etc.
```

File: contexts/credentials/v2
Comment (TallTed) [reply contexts/credentials/v2:66]: By "scoped to any property/ies" above, I meant that this `confidenceMethod` should be applicable to any property+value pair (or group of same), something like 
```
_:property
   [ _:value          12345 ; 
     confidenceMethod `{uri to method used to get or usable to test value}` 
   ]
```

**Comment by TallTed** [comment]
@OR13 -- `Update reservation table` is accurate but uninformative. Please change the title of this PR to reflect what those updates are, something like ```Add confidenceMethod to table of reserved terms and vc `@context` v2```

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-06-27-vcwg-special#section2) on 2023-06-27

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 2. Update reservation table (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Brent Zundel:** 1142 updates the reservation table with confirmation method and render method. Render method's URLs work. confirmation URL dont work yet.  

> *Orie Steele:* It is blocked pending adoption of the work item, which was JUST proposed to the CCG.

**Brent Zundel:** as soon as that is officially adopted in the CCG the URLs will work and I will approve.  

**Kristina Yasuda:** Is there a timeline for that?  

> *Orie Steele:* CCG already has 2 companies supporting it, it is waiting for acknowledgement by chairs, it cannot be blocked.

**Oliver Terbu:** We just created the new work item proposal today. others can give more details on timeline.  

> *Orie Steele:* Nobody can stop a CCG work item from being worked on when multiple companies support it.

**Michael Prorock:** Harrison is chairing the meeting today and will note. Send a message to the chairs and we can expedite it. If you have multiple parties and there are no objections, it will go through.  

**Kristina Yasuda:** If you want to follow the confidence method work, pay attention to the CCG.  

**Oliver Terbu:** We are still looking for owners of the work item.  

**Kristina Yasuda:** if you are interested in moving this forward,contact the CCG or reach out to oliver.  

**Orie Steele:** Confused. I though spruce and digital bazaar are sponsoring this work?  

**Oliver Terbu:** We don't have the bandwidth to own the work. Didn't say we would not implement.  

**Kristina Yasuda:** lets take these questions offline.  

> *Michael Prorock:* +1.

**Manu Sporny:** Digital bazaar is supportive of the work, but currently we don't have the bandwidth to be an owner of the work. We see value in it. cant work on it for 6 mo or so.  

**Kristina Yasuda:** Any coordinate needed with CCG?  

> *Orie Steele:* +1 to dropping it, given that nobody wants to work on it in ccg.

**Michael Prorock:** If there is no one working on it, can we drop it from the VCWG side? On the CCG side we need two owners from different orgs to approve.  

**Kristina Yasuda:** Lets wait and see.  

</details>

File: index.html
Comment (brentzundel) [reply index.html:3307]: @awoie this link is still 404

File: index.html
Comment (OR13) [reply index.html:3298]: Seems odd to remove holder here? 

It's part of the 3 part model, isn't it important to highlight that here?

Comment (TallTed) [reply index.html:3298]: Inadvertent omission in the rewritten sentence. Fixed.

**Comment by OR13** [comment]
So far, nobody has stepped forward to work on `confidenceMethod` as W3C CCG... https://github.com/w3c-ccg/community/issues/245#issuecomment-1622029409

I suggest we remove the registration request unless that changes.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-05-vcwg#section1-1) on 2023-07-05

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. Add confidenceMethod to table of reserved terms and v2 `@context` (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Michael Prorock:** If no one is stepping up to do #1142 maybe we can close it and pick it up later if need be.  

> *Manu Sporny:* I'd be a +1 to close the PR, but we should link to specs that exist.

> *Manu Sporny:* agree that we shouldn't close w/o Oliver being here.

**Kristina Yasuda:** Oliver's not on the call so we can't close now.  

**Michael Prorock:** I didn't mean close right now. I just don't want it to hang out indefinitely.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-11-vcwg-special#section1-3) on 2023-07-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.3. Add confidenceMethod to table of reserved terms and v2 `@context` (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Brent Zundel:** this next one is around adding confidence method as a reserved term.  
… current understanding is this is blocked awaiting adoption from the CCG group to move this work forward.  

> *Orie Steele:* Per the last comment on it, I will remove the confidenceMethod part in a week or 2 if there is still not CCG adoption.

**Oliver Terbu:** currently waiting for owners of this work in the CCG. Nobody has stood up to take on this work.  
… intend to wait two weeks, if none willing to take this on then we should close this P.R in the VCDM.  

**Brent Zundel:** started 2 week time out 5 days ago.  

**Dave Longley:** question around why 2 weeks timeout as opposed to just before CR.  

**Oliver Terbu:** I could be supportive of this.  

**Brent Zundel:** goal of the shorter timeline is to not have P.Rs hanging around indefinitely.  
… want P.Rs to be progressing, merging or closing.  

**Oliver Terbu:** If we don't find anyone in two weeks, then we could create an issue to track this work.  
… Need one person to be the main driver of this spec in the CCG.  

> *Dave Longley:* +1 to the issue compromise approach, clearly there's support for the work just people too busy right now.

**Oliver Terbu:** have multiple copilots, but need a main driver that I can coordinate with.  

**Michael Jones:** generally have policy to only put normative things in the spec.  
… CCG work item does not qualify to add an identifier to our spec.  
… CCG is not standards track.  

> *Orie Steele:* +1 selfissued, but the registration table is a cheat code :).

**Brent Zundel:** recognize that, but this is about getting into the table of reserved properties.  
… expect selfissued concerns to be addressed before CR.  
… moving on.  

</details>

**Comment by awoie** [comment]
Three owners have been found now: https://github.com/w3c-ccg/community/issues/245

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-11-vcwg-special#section1-3) on 2023-07-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.3. Add confidenceMethod to table of reserved terms and v2 `@context` (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Brent Zundel:** this next one is around adding confidence method as a reserved term.  
… current understanding is this is blocked awaiting adoption from the CCG group to move this work forward.  

> *Orie Steele:* Per the last comment on it, I will remove the confidenceMethod part in a week or 2 if there is still not CCG adoption.

**Oliver Terbu:** currently waiting for owners of this work in the CCG. Nobody has stood up to take on this work.  
… intend to wait two weeks, if none willing to take this on then we should close this P.R in the VCDM.  

**Brent Zundel:** started 2 week time out 5 days ago.  

**Dave Longley:** question around why 2 weeks timeout as opposed to just before CR.  

**Oliver Terbu:** I could be supportive of this.  

**Brent Zundel:** goal of the shorter timeline is to not have P.Rs hanging around indefinitely.  
… want P.Rs to be progressing, merging or closing.  

**Oliver Terbu:** If we don't find anyone in two weeks, then we could create an issue to track this work.  
… Need one person to be the main driver of this spec in the CCG.  

> *Dave Longley:* +1 to the issue compromise approach, clearly there's support for the work just people too busy right now.

**Oliver Terbu:** have multiple copilots, but need a main driver that I can coordinate with.  

**Michael Jones:** generally have policy to only put normative things in the spec.  
… CCG work item does not qualify to add an identifier to our spec.  
… CCG is not standards track.  

> *Orie Steele:* +1 selfissued, but the registration table is a cheat code :).

**Brent Zundel:** recognize that, but this is about getting into the table of reserved properties.  
… expect selfissued concerns to be addressed before CR.  
… moving on.  

</details>

**Comment by brentzundel** [comment]
> Three owners have been found now: [w3c-ccg/community#245](https://github.com/w3c-ccg/community/issues/245)

Excellent. Once the URLs resolve properly, I believe we can move forward with this PR.

**Comment by msporny** [comment]
> > Three owners have been found now: [w3c-ccg/community#245](https://github.com/w3c-ccg/community/issues/245)
> 
> Excellent. Once the URLs resolve properly, I believe we can move forward with this PR.

@mprorock, @man4prez, @kwlinson -- Can you please provide a final determination on whether or not the Confidence Method Extension specification has been adopted by CCG? It has met all of the work item adoption criteria, AFAICT.

That determination is blocking this PR in VCWG.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-26-vcwg#section1-1) on 2023-07-26

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. Add confidenceMethod to table of reserved terms and v2 `@context` (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Manu Sporny:** 1142 is about confidence method. Progress has been made on it; we should be able to merge the pull request shortly.  

</details>

**Comment by msporny** [comment]
The work item has been approved at this point: https://github.com/w3c-ccg/community/issues/245#issuecomment-1650369356

We are waiting for the repo to be moved to CCG, so the URLs resolve, so we can merge this PR.

File: index.html
Comment (msporny) [inline index.html:3307]:
```suggestion
See <a href="https://w3c-ccg.github.io/confidence-method-spec/">Verifiable Credential Confidence Methods</a>.
```

Comment (msporny) [reply index.html:3307]: This URL now resolves, once the change request is merged, we can merge the PR.

Comment (TallTed) [reply index.html:3307]: https://w3c-ccg.github.io/vc-confidence-method/ is still 404

Comment (TallTed) [reply index.html:3307]: will be fixed by @msporny's https://github.com/w3c/vc-data-model/pull/1142/files#r1278134776

**Review by brentzundel — APPROVED** [review]

**Review by shigeya — APPROVED** [review]

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-01-vcwg-special#section1-3) on 2023-08-01

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.3. Add confidenceMethod to table of reserved terms and v2 `@context` (pr vc-data-model#1142)

_See github pull request [vc-data-model#1142](https://github.com/w3c/vc-data-model/pull/1142)._




**Brent Zundel:** this one's 1142, Add ConfidenceMethod to table of reserved terms.  
… it was being held up till the URLs actually resolved, I believe they do now.  
… any reason not to just merge?  
… (not the PR, the requested changes).  
… Manu has one that fixes the URL, TallTed has another one that.  

**Manu Sporny:** I was going to suggest, this was before Ted's change, if we could just merge both of these suggestions, we can merge the PR.  
… we're waiting on Orie to process them.  

**Ted Thibodeau Jr.:** that's not all, there is - in a Jun 6 comment, there's a URL to vc confidence method that's still 404.  

**Manu Sporny:** I did a change suggestion to that, to fix it.  

**Brent Zundel:** ok, I'm very confident that we can add Manu's change suggestion.  
… and roughly confident that we can add Ted's. waiting on official thumbs-up from Orie.  
… but also don't want this PR blocked.  
… manu, will you take an action to merge your change suggestion, fixing the PR? I can change my request into an approval.  
… and we can ping Orie to look at Ted's suggestions.  

**Manu Sporny:** I've merged my URL change.  

**Brent Zundel:** excellent, I marked PR as approved.  
… any folks on the call who have yet to review any of these PRs, please do so.  
… if you don't, things may get merged that you haven't had a chance to review -- we need to be brisk now.  
… moving on, next one is 1212.  

</details>

**Comment by brentzundel** [comment]
Multiple positive review, open for more than a week. Changes requested and made. Merging.

File: index.html
Comment (TallTed) [inline index.html:3300]: I know I missed the merge. You can probably fix this with a direct commit, rather than going through all the PR hoops...
```suggestion
limited to attributes such as `initialRecipient` (a/k/a `issuee`), `presenter`,
```

Comment (brentzundel) [reply index.html:3300]: Thanks for catching this. Fixed in [cf429b1](https://github.com/w3c/vc-data-model/commit/cf429b194ff89ae255a70cf271aac45f703f96c1)

**Pull request merged**
